lightning/src/util/sweep.rs

```diff
@@ -472,7 +472,19 @@ where
 			return Ok(());
 		}
 
-		let result = self.regenerate_and_broadcast_spend_if_necessary_internal().await;
+		let result = {
+			self.regenerate_and_broadcast_spend_if_necessary_internal().await?;
+
+			// If there is still dirty state, we need to persist it.
+			let mut sweeper_state = self.sweeper_state.lock().unwrap();
+			if sweeper_state.dirty {
+				self.persist_state(&mut *sweeper_state).map_err(|e| {
```

Comment:
Would rather take a non-mutable reference here and update the `dirty` flag afterwards, since `persist_state` seems like it should be "read-only" on the state. Or we could rename the method.

Reply:
Renamed